lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
+        """Returns a matrix designed to contain suitable regressors for the
+        systematics noise model.
+
+        The design matrix contains one row for each cadence (i.e. moment in time)
+        and one column for each regressor that we wish to use to predict the
+        systematic noise in a given cadence.
+
+        The columns (i.e. regressors) included in the design matrix are:
+        * One column for each pixel in the PLD aperture mask.  Each column
+          contains the flux values observed by that pixel over time.  This is
+          also known as the first order component.
+        * Columns derived from the products of all combinations of pixel values
+          in the aperture mask. However, rather than including a column for each
+          combination, we perform dimensionality reduction (PCA) and include a
+          smaller number of PCA terms, i.e. the number of columns is
+          n_pca_terms*(pld_order-1).  This is also known as the higher order
+          components.
+        * Optionally, a single column of ones for numerical stability.
+
+        Thus, the shape of the design matrix will be
+        (n_cadences, n_pld_mask_pixels + n_pca_terms*(pld_order-1) + include_column_of_ones)
 
-    def correct(self, aperture_mask=None, cadence_mask=None, gp_timescale=30,
-                use_gp=True, pld_order=2, n_pca_terms=10, pld_aperture_mask=None):
-        r"""Returns a PLD systematics-corrected LightCurve.
+        Returns
+        -------
+        design_matrix : 2D numpy array
+            See description above.
+        """
+        # We use an optional dependency for very fast PCA (fbpca), but if the
+        # import fails we will fall back on using the slower `np.linalg.svd`.
+        use_fbpca = True
+        try:
+            from fbpca import pca
+        except ImportError:
+            use_fbpca = False
+            log.warning("PLD systematics correction will run faster if the "
+                        "optional `fbpca` package is installed "
+                        "(`pip install fbpca`).")
+
+        matrix_sections = []  # list to hold the design matrix components
+        first_order_matrix = self.create_first_order_matrix()
+
+        # Input validation: n_pca_terms cannot be larger than the number of cadences
+        n_cadences = len(first_order_matrix)
+        if n_pca_terms > n_cadences:
+            log.warning("`n_pca_terms` ({}) cannot be larger than the number of cadences ({});"
+                        "using n_pca_terms={}".format(n_pca_terms, n_cadences, n_cadences))
+            n_pca_terms = n_cadences
+
+        # The original EVEREST paper includes a column vector of ones in the
+        # design matrix to improve the numerical stability (see Luger et al.);
+        # it is unclear whether this is necessary, so this is an optional step
+        # for now.
+        if include_column_of_ones:
+            matrix_sections.append([np.ones((len(first_order_matrix), 1))])
```

Comment:
Why isn't the array of ones added as the last step? Why is it 1st order, column of ones, 2nd order?

Reply:
After some experimentation this afternoon, it seems like this step is unnecessary, so I've removed the column of ones. I think this is because we're simultaneously fitting the GP and the mean along with the PLD coefficients, which removes the need to include the assumption that the light curve is otherwise "flat" without the GP trend.

Reply:
Ok, I take this back. I found a case where the column of ones prevents a crash:

When the design matrix is small in both dimensions, meaning few cadences (<~500) and few column vectors (<~25). One of the tests fit this description and started failing when I removed the column of ones. 

The error that Theano raises is very opaque, so I've created a `LightkurveError` class and raise that with a more informative warning in this case. @barentsen please weigh in on this choice (see the `optimize` call in the `correct` function).

I'm putting the option to include the column of ones back in, but as the last column (though this doesn't seem to make a difference).